### PR TITLE
Build and Package `cuvs_bench` in CI

### PR DESCRIPTION
**Pull request by divyegala**
This PR sets up building and packaging of `cuvs_bench` in CI, which will automatically setup publishing of `cuvs_bench` packages to rapids conda channels

### CODE REVIEW

File: conda/recipes/cuvs_bench/meta.yaml
Comment (cjnolet) [inline conda/recipes/cuvs_bench/meta.yaml:93]: One thing that's been a pain in the RAFT benchmarks is that we weren't installin gthese things in the developer environment. Can we add these to a special developer environment in the dependencies.yaml so that users who want to build the benchmarks from source can have these installed automatically?

Comment (divyegala) [reply conda/recipes/cuvs_bench/meta.yaml:93]: They are there in the dev env https://github.com/rapidsai/cuvs/blob/f39b8d512355bb46a247b1591bcb3f4725e5a85c/conda/environments/bench_ann_cuda-118_arch-x86_64.yaml#L25-L26

**Comment by cjnolet** [comment]
I see some c++-level cusolver include build errors in the Python package.

**Review by cjnolet — APPROVED** [review]

File: cpp/cmake/thirdparty/get_cuvs.cmake
Comment (cjnolet) [inline cpp/cmake/thirdparty/get_cuvs.cmake:2]: Please use only current year for all new files.

**Review by cjnolet — APPROVED** [review]
Approving, but please update the copyright for new files to only the current year. The changes here look good.

File: build.sh
Comment (cjnolet) [inline build.sh:42]: RAFT needs to be changed to cuVS everywhere.

**Review by cjnolet — CHANGES_REQUESTED** [review]
Sorry, Divye. I noticed in your other PR that there are many places RAFT still needs to be change dto cuVS. I can't approve this yet. However, these should be minor and simple changes to make. Shall we use your other PR going forward and close this one to make things more manageable?

**Review by bdice — APPROVED** [review]

**Comment by bdice** [comment]
Approving, should be fine to merge once the RAFT/cuVS name changes are figured out.

**Review by AyodeAwe — APPROVED** [review]

**Comment by cjnolet** [comment]
Closing for now because changes are in #368. Please open back up if this is absolutely neeed.

**Pull request closed without merging**
